experimental/packages/opentelemetry-sdk-metrics/test/export/ConsoleMetricExporter.test.ts

```diff
@@ -38,71 +44,95 @@ async function waitForNumberOfExports(exporter: sinon.SinonSpy<[metrics: Resourc
-    meterReader = new PeriodicExportingMetricReader({
-      exporter: exporter,
-      exportIntervalMillis: 100,
-      exportTimeoutMillis: 100
+  describe('export', () => {
+    let previousConsoleDir: any;
+    let exporter: ConsoleMetricExporter;
+    let meterProvider: MeterProvider;
+    let meterReader: PeriodicExportingMetricReader;
+    let meter: metrics.Meter;
+
+    beforeEach(() => {
+      previousConsoleDir = console.dir;
+      console.dir = () => {};
+
+      exporter = new ConsoleMetricExporter();
+      meterProvider = new MeterProvider({ resource: defaultResource });
+      meter = meterProvider.getMeter('ConsoleMetricExporter', '1.0.0');
+      meterReader = new PeriodicExportingMetricReader({
+        exporter: exporter,
+        exportIntervalMillis: 100,
+        exportTimeoutMillis: 100
+      });
+      meterProvider.addMetricReader(meterReader);
     });
-    meterProvider.addMetricReader(meterReader);
-  });
 
-  afterEach(async () => {
-    console.dir = previousConsoleDir;
+    afterEach(async () => {
+      console.dir = previousConsoleDir;
+
+      await meterReader.shutdown();
+    });
+
+    it('should export information about metric', async () => {
+      const counter = meter.createCounter('counter_total', {
+        description: 'a test description',
+      });
+      const counterAttribute = { key1: 'attributeValue1' };
+      counter.add(10, counterAttribute);
+      counter.add(10, counterAttribute);
+
+      const histogram = meter.createHistogram('histogram', { description: 'a histogram' });
+      histogram.record(10);
+      histogram.record(100);
+      histogram.record(1000);
+
+      const spyConsole = sinon.spy(console, 'dir');
+      const spyExport = sinon.spy(exporter, 'export');
+
+      await waitForNumberOfExports(spyExport, 1);
+      const resourceMetrics = spyExport.args[0];
+      const firstResourceMetric = resourceMetrics[0];
+      const consoleArgs = spyConsole.args[0];
+      const consoleMetric = consoleArgs[0];
+      const keys = Object.keys(consoleMetric).sort().join(',');
 
-    await meterReader.shutdown();
+      const expectedKeys = [
+        'dataPointType',
+        'dataPoints',
+        'descriptor',
+      ].join(',');
+
+      assert.ok(firstResourceMetric.resource.attributes.resourceKey === 'my-resource', 'resourceKey');
+      assert.ok(keys === expectedKeys, 'expectedKeys');
+      assert.ok(consoleMetric.descriptor.name === 'counter_total', 'name');
+      assert.ok(consoleMetric.descriptor.description === 'a test description', 'description');
+      assert.ok(consoleMetric.descriptor.type === 'COUNTER', 'type');
+      assert.ok(consoleMetric.descriptor.unit === '', 'unit');
+      assert.ok(consoleMetric.descriptor.valueType === 1, 'valueType');
+      assert.ok(consoleMetric.dataPoints[0].attributes.key1 === 'attributeValue1', 'ensure metric attributes exists');
+
+      assert.ok(spyExport.calledOnce);
+    });
```

Comment:
This code here is not new and has only been moved into a new describe block, as the setup is not needed for the tests that I added.